### ISSUE TITLE
Ironisight is correct and adjusted trough SWEP.IronSights but optics are misaligned

### ISSUE THREAD

**Issue by mteomi** [HaodongMo/ARC-9#251]
Basically I managed to fix my Ironsight so that they could be aligned with the way you suggested me and It worked, but now whenever I put an optic (Bone = "tag_scope", Pos = Vector(1, 0, -0.1), Ang = Angle(0, 0, 0),) It just isn't correctly allocated, I put a screenshot of my Blender in case it has something to do with how and where the bone is positioned.

![bone](https://github.com/HaodongMo/ARC-9/assets/97291968/33d585a8-0c4b-451b-ada9-8740aa1a7445)

I tried guessing some solutions, but the bone seems correctly positioned to me.

Showcase: https://youtu.be/YTHHhOQrnsY

**Comment by CurlySparkle** [comment]
If the attachment it self is attached correctly on the weapon but the sight is bad then you can try using CorrectiveAng on the SWEP.Attachments to try to see if this fixes the misaligned, if that doesn't help much then you can try checking on the ATT.Ironisight of the sight attachment it self (in this mostly if you are making a new sight att)

**Comment by mteomi** [comment]
I tried CorrectiveAng and it doesn't do anything, no difference when I check, I tried to restart the map, same result, I tried different sights, nothing.
Could it be that the bone is slightly misaligned? Just guessing, never deleted any bones, just moved a little bit tag_scope. I tried different bones: Bone = "tag_reflex", Bone = "tag_holo", maybe it's also a bone Issue.

Now the red dot is aligned after I've set Ang = Angle(-0.6, -0.6, 0),  but not the sight itself, it's better to show than to describe it with words:

https://youtu.be/ajzNeqkGDSU

![20240620012837_1](https://github.com/HaodongMo/ARC-9/assets/97291968/5464a4ef-a1d4-46fb-a7e2-f89773984739)
![20240620012849_1](https://github.com/HaodongMo/ARC-9/assets/97291968/856dbde1-22fc-45d9-83bb-533a5b4469c6)

**Comment by CurlySparkle** [comment]
Yeah this is mostly because the animation is forcing the ADS position, Like darsu explained before setting stuff like this is complicated specially if you try to use poseparams, i would go mostly using the SWEP.Ironsight method instead with no ADS in/out, you can still use other anims for the ADS tho like fire_ads or reloads with ADS option but youll need to combine them in with the normal animations like this
![image](https://github.com/HaodongMo/ARC-9/assets/5094277/3738bc9d-2252-4f01-8044-a81ab43f9d26)
where the line of blend is is the combination to make both anims work one for hip and the other when you are on ADS

**Comment by mteomi** [comment]
About that: I checked and I am sure there is no poseparams anymore, no ADS in/out, I removed the references from the lua file and from the QC file.

**Comment by CurlySparkle** [comment]
if the ADS anim aren't playing then must be the default pose supports that ARC9 has are currently broken, ether way if you also trying to setup this weapon specifically from a pack you can use their setup that they have in this case since the weapon is the BP50 from vanguard you can use the setup calls like i showed on the QC, for the problem with the sight position not assigning correctly then must be something on the lua that's forcing it

**Comment by mteomi** [comment]
> for the problem with the sight position not assigning correctly then must be something on the lua that's forcing it
Now that I think the lua file comes from arc9_mw22_lachman762.lua, so maybe there is some parameter that is causing this I guess?

**Comment by CurlySparkle** [comment]
not sure but i don't think so since most of these weapons don't use ads anims, they only use the blending that's inside the QC but not on the lua to use them separately
